src/workerd/util/mimetype.c++

```diff
@@ -6,24 +6,64 @@
 #include <kj/debug.h>
 #include <kj/string-tree.h>
 #include <workerd/util/string-buffer.h>
+#include <array>
 
 namespace workerd {
 
 namespace {
 
-bool isWhitespace(const char c) {
+constexpr bool isWhitespace(const char c) noexcept {
   return (c == '\r' || c == '\n' || c == '\t' || c == ' ');
 }
 
-bool isTokenChar(const unsigned char c) {
-  return c == '!' || c == '#' || c == '$' || c == '%' || c == '&' || c == '\'' ||
-         c == '*' || c == '+' || c == '\\' || c == '-' || c == '.' || c == '^' ||
-         c == '_' || c == '`' || c == '|' || c == '~' || (c >= 'A' && c <= 'Z') ||
-         (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+constexpr std::array<uint8_t, 256> token_table = []() constexpr {
+  std::array<uint8_t, 256> result{};
+
+  for (uint8_t c : { '!' ,  '#' ,  '$' ,  '%' ,  '&' ,  '\'' ,
+          '*' ,  '+' ,  '\\' ,  '-' ,  '.' ,  '^' ,
+          '_' ,  '`' ,  '|' ,  '~'}) {
+    result[c] = true;
+  }
+
+  // (c >= 'A' && c <= 'Z')
+  for (uint8_t c = 65; c <= 90; c++) {
+    result[c] = true;
+  }
+
+  // (c >= 'a' && c <= 'z')
+  for (uint8_t c = 97; c <= 122; c++) {
+    result[c] = true;
+  }
+
+  // (c >= '0' && c <= '9')
+  for (uint8_t c = 48; c <= 57; c++) {
+    result[c] = true;
+  }
+
+  return result;
+}();
+
+constexpr bool isTokenChar(const uint8_t c) noexcept {
+  return token_table[c];
 }
 
-bool isQuotedStringTokenChar(const unsigned char c) {
-  return (c == '\t') || (c >= 0x20 && c <= 0x7e) || (c >= 0x80);
+constexpr std::array<uint8_t, 256> quoted_string_token_table = []() constexpr {
```

Comment:
Do we want this lambda to be consteval instead? just to be super duper sure it's evaluated at compiletime?

Reply:
Likewise on the token_table lambda